cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -131,11 +131,13 @@ func SetClusterDynamicDefaults(cfg *kubeadmapi.ClusterConfiguration, advertiseAd
 	// Default all the embedded ComponentConfig structs
 	componentconfigs.Known.Default(cfg)
 
-	ip := net.ParseIP(advertiseAddress)
-	if ip.To4() != nil {
-		cfg.ComponentConfigs.KubeProxy.BindAddress = kubeadmapiv1beta2.DefaultProxyBindAddressv4
-	} else {
-		cfg.ComponentConfigs.KubeProxy.BindAddress = kubeadmapiv1beta2.DefaultProxyBindAddressv6
+	if cfg.ComponentConfigs.KubeProxy != nil {
```

Comment:
Ummm, I don't think this is necessary. @neolit123 
`cfg.ComponentConfigs.KubeProxy` is never `nil` after `componentconfigs.Known.Default(cfg)` executes.